exporter/awss3exporter/marshaler.go

```diff
@@ -23,6 +25,20 @@ var (
 	ErrUnknownMarshaler = errors.New("unknown marshaler")
 )
 
+func newMarshalerFromEncoding(encoding *component.ID, host component.Host, logger *zap.Logger) (marshaler, error) {
+	marshaler := &s3Marshaler{logger: logger}
+	e, ok := host.GetExtensions()[*encoding]
+	if !ok {
+		return nil, fmt.Errorf("unknown encoding %q", encoding)
+	}
+	// cast with ok to avoid panics.
+	marshaler.logsMarshaler, _ = e.(plog.Marshaler)
+	marshaler.metricsMarshaler, _ = e.(pmetric.Marshaler)
+	marshaler.tracesMarshaler, _ = e.(ptrace.Marshaler)
+	marshaler.fileFormat = encoding.String()
```

Comment:
@atoulme - what's the expected behaviour here, is this intended to be the ID as defined in the config file?

```yaml
extension:
   textencoding: # <- fileFormat == "" ?
   textencoding/myspecialtextencoding: # <- fileFormat == "myspecialtextencoding"?
```

Reply:
The whole ID: textencoding/myspecialtextencoding

Reply:
Thanks for looking into this. 1 and 2 seem like the right idea to me, with the extension by default being an empty string. Do you want to file an issue to follow up?

Reply:
I've raised #31818